R/utils_examples.R

```diff
@@ -0,0 +1,83 @@
+#' Utilities to write and read tables in html for function documentation
+#'
+#' This is still experimental. We are planning to potentially expand this to other
+#' html output. Principally based on [as_gt()] and [gt::as_raw_html()].
+#'
+#' @param tbl (`gtsummary` or `html`)\cr tables are converted to html. Any html code (table or not)
+#'   can be saved and retrieved for documentation purposes.
+#' @param example_name (`string`)\cr usually the name of the function or file where the
+#'   table is used.
+#' @param out_var_name (`string`)\cr the name of the variable that will be saved. It defaults
+#'   to the name of the variable passed to the function (`tbl`).
+#' @param method (`string`)\cr saving method. Default is `"rds"` as it produces a smaller file,
+#'   but we are planning to extend this to other, non-html outputs like `"png"`.
+#'
+#' @keywords internal
+#' @name utils_examples
+write_example_output <- function(tbl,
+                                 example_name = "example",
+                                 out_var_name = deparse(substitute(tbl)),
+                                 method = c("txt", "rds")[2]) {
+  if (isFALSE(method %in% c("txt", "rds"))) {
+    stop("Inserted method is not supported.")
+  }
+
+  if (getOption("gtsummary_update_examples", default = FALSE)) {
+    if (is(tbl, "html")) {
+      html_output <- tbl
+    } else if (is(tbl, "gtsummary")) {
+      html_output <- tbl |>
+        as_gt() |>
+        gt::tab_options() |>
+        gt::as_raw_html()
+    } else {
+      stop("tbl needs a html or gtsummary table. Got something else.")
+    }
```

Comment:
In some of the documentation, we already convert gtsummary tables to gt.

Perhaps this section could look like 

```r
if (is(tbl, "gtsummary") <convert to gt with as_gt()>
if (is(tbl, "gt_tbl")) <do the html magic>
```

Reply:
done ;)